src/librustc_ast_passes/ast_validation.rs

```diff
@@ -49,6 +49,13 @@ impl BoundContext {
 
 struct AstValidator<'a> {
     session: &'a Session,
+
+    /// The span of the `extern` in an `extern { ... }` block, if any.
+    extern_mod_span: Option<Span>,
```

Comment:
```suggestion
    extern_mod: Option<&'a Item>,
```
It would be cheaper to retrieve the span only when the error is actually reported.